llog/glog.go

```diff
@@ -1159,3 +1160,48 @@ func (l *Log) V(level Level) bool {
 func (l *Log) Stats() Stats {
 	return *l.stats
 }
+
+// CopyStandardLogTo arranges for messages written to the Go "log" package's
+// default logs to also appear in the Google logs for the named and lower
+// severities.
+func (l *Log) CopyStandardLogTo(name string) {
+	sev, ok := severityByName(name)
+	if !ok {
+		panic(fmt.Sprintf("log.CopyStandardLogTo(%q): unrecognized severity name", name))
+	}
+	// Set a log format that captures the user's file and line:
+	//   d.go:23: message
+	stdLog.SetFlags(stdLog.Lshortfile)
+	stdLog.SetOutput(logBridge{severity: sev, log: l})
+}
+
+// logBridge provides the Write method that enables CopyStandardLogTo to connect
+// Go's standard logs to the logs provided by this package.
+type logBridge struct {
+	severity Severity
+	log      *Log
+}
+
+// Write parses the standard logging line and passes its components to the
+// logger for severity(lb).
+func (lb logBridge) Write(b []byte) (n int, err error) {
+	var (
+		file = "???"
+		line = 1
+	)
+	buf := lb.log.getBuffer()
+	// Split "d.go:23: message" into "d.go", "23", and "message".
+	if parts := bytes.SplitN(b, []byte{':'}, 3); len(parts) != 3 || len(parts[0]) < 1 || len(parts[2]) < 1 {
+		fmt.Fprint(buf, fmt.Sprintf("bad log format: %s", b))
+	} else {
+		file = string(parts[0])
+		line, err = strconv.Atoi(string(parts[1]))
+		fmt.Fprint(buf, string(b))
+		if err != nil {
+			fmt.Fprint(buf, fmt.Sprintf("bad line number: %s", b))
+			line = 1
+		}
+	}
+	lb.log.output(Severity(lb.severity), buf, file, line)
```

Comment:
Is there a reason to not prepend the headers prior to output? (as all other callees of output do).

This results in the logBridge lines being inconsistent with the log format (namely, just filename:lineno msg) get written?

Seems like you'd want a custom variant of the header function that takes in the file/line and then you append parts[2] to the buffer.